delivery_price_rule_untaxed/models/delivery_carrier.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+from odoo.tools.safe_eval import safe_eval
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = 'delivery.carrier'
+
+    @api.multi
+    def get_price_available(self, order):
+        self.ensure_one()
+        self = self.with_context(
+            {'order_amount_untaxed': order.amount_untaxed})
+        res = super(DeliveryCarrier, self).get_price_available(order)
+        return res
+
+    def get_price_from_picking(self, total, weight, volume, quantity):
+        price = 0.0
+        criteria_found = False
+        price_dict = {
+            'price': total,
+            'volume': volume,
+            'weight': weight,
+            'wv': volume * weight,
+            'quantity': quantity,
+            'untaxed_price': self.env.context.get('order_amount_untaxed', 0)
+        }
+        for line in self.price_rule_ids:
+            test = safe_eval(
+                line.variable + line.operator + str(line.max_value),
+                price_dict)
+            if (test and 'untaxed_price' in
+                    (line.variable, line.variable_factor)):
+                price = (line.list_base_price
+                         + line.list_price * price_dict[line.variable_factor])
+                criteria_found = True
+                break
+            elif test:
+                return super(DeliveryCarrier, self) \
+                    .get_price_from_picking(total, weight, volume, quantity)
+        if not criteria_found:
```

Comment:
I think you should call super outside the loop and you don't need this extra `raise`

Reply:
Done in https://github.com/OCA/delivery-carrier/pull/168/commits/ab383aaafe228f765fd393fd51a03b0f0fb7bcc4